torch/_weights_only_unpickler.py

```diff
@@ -23,6 +23,7 @@
 # weights = torch.load(buf, weights_only = True)
 
 import functools as _functools
+from _compat_pickle import IMPORT_MAPPING, NAME_MAPPING
```

Comment:
Have a comment that this is privte cpython.
try/catch to handle when it's missing (nice error message)
And make it is available for all versions we use.

Reply:
To avoid warning on `import torch`, I'm warning only when `Unpickler.load` is called, does this sound reasonable to you?